snapcraft/internal/cache/_cache.py

```diff
@@ -33,13 +33,14 @@ def __init__(self):
     def cache(self):
         raise NotImplementedError
 
-    def prune(self):
+    def prune(self, revision=None):
```

Comment:
`revision` shouldn't be in `SnapcraftCache`, if necessary go with the `**kwargs, *args` trick here for the cases we need to call `super`